src/AggregateFunctions/AggregateFunctionFunnelAnalysis.h

```diff
@@ -0,0 +1,288 @@
+#pragma once
+
+#include <iostream>
+#include <sstream>
+#include <vector>
+#include <unordered_set>
+#include <map>
+#include <Columns/ColumnsNumber.h>
+#include <DataTypes/DataTypeDateTime.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <IO/ReadHelpers.h>
+#include <IO/WriteHelpers.h>
+#include <Columns/ColumnArray.h>
+#include <Columns/ColumnString.h>
+#include <Common/ArenaAllocator.h>
+#include <Common/assert_cast.h>
+#include <common/logger_useful.h>
+#include <Poco/Logger.h>
+
+#include <AggregateFunctions/AggregateFunctionNull.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int BAD_ARGUMENTS;
+}
+
+struct ComparePair final
```

Comment:
We have same struct in AggregateFunctionWindowFunnel.h. Maybe make a header with it and include it in both files to avoid code duplication? Maybe smth like Common/ComparePair.h